Python/chapter03/p03_stack_of_plates/miguelHx.py

```diff
@@ -130,23 +152,11 @@ def __str__(self):
         return '->'.join(values)
 
 
-class SetOfStacksIterator(Generic[T]):
-    def __init__(self, stack_list: List[MyStack[T]]):
-        # build temporary list
-        self.my_list = []
-        self.index = 0
-        stack_list_reversed = stack_list[::-1]
-        for stack in stack_list_reversed:
-            for item in stack:
-                self.my_list.append(item)
-
-    def __next__(self) -> T:
-        if self.index == len(self.my_list):
-            raise StopIteration
-        item: T = self.my_list[self.index]
-        self.index += 1
-        return item
-
+def yield_set_of_stacks(stack_list: List[MyStack[T]]):
```

Comment:
per https://docs.python.org/3/library/typing.html#typing.Generator :

```suggestion
def yield_set_of_stacks(stack_list: List[MyStack[T]]) -> Generator[T, None, None]:
```

or

```suggestion
def yield_set_of_stacks(stack_list: List[MyStack[T]]) -> Iterator[T]:
```

Reply:
8b2e4d4fb0fbd24108854057af32a8078ad2e9d8